pkg/sql/sem/tree/backup.go

```diff
@@ -67,8 +72,9 @@ func (node *Backup) Format(ctx *FmtCtx) {
 	}
 	if node.Nested {
 		ctx.WriteString("INTO ")
-		if node.AppendToLatest {
```

Comment:
do we not still need this?

Reply:
node.Subdir will never be nil if node.Nested is true. It'll be set in backup planning as long as node.Nested is true. Is there a case we want to still print `LATEST IN` which i'm missing?

Reply:
what does this case in sql.y do? https://github.com/cockroachdb/cockroach/pull/53139/files#diff-15c44669e939ef2a26ac13bed5a93009R2130

Reply:
`BACKUP opt_backup_targets INTO LATEST IN string_or_placeholder_opt_list...`

becomes:

`BACKUP INTO 'resolved-LATEST-subdir' IN string_or_placeholder_opt_list...` because of:
```
if b.Nested {
		b.Subdir = tree.NewDString(resolvedSubdir)
}
``` 
in `backupJobDescription()`. Where resolvedSubdir is the `chosenSuffix` in backup planning.

Reply:
Oh, I think I see what you mean: in the tree.Backup _you_ create in `backupJobDesc` it'll never be nil since we'll resolve latest to a value, but we have to be able to round-trip any AST node to/from SQL, even before planning. I'm pretty sure you want `if node.AppendToLatest { .. else } if node.Subdir != nil { ...}`

Reply:
got it, wasn't aware we need to be able to round-trip. will change it!